monix-reactive/shared/src/test/scala/monix/reactive/consumers/ForeachAsyncConsumerSuite.scala

```diff
@@ -101,4 +101,21 @@ object ForeachAsyncConsumerSuite extends TestSuite[TestScheduler] {
     s.tick()
     assert(cancelled)
   }
+
+  test("should suspend effects encountered during the stream") { implicit s =>
+    val exceptionMessage: String = "Boom!"
+    val t: Try[Unit] = Try {
+      Observable(1)
+        .consumeWith(
+          Consumer.foreachTask(_ =>
+            Task.raiseError(new RuntimeException(exceptionMessage))
+          )
+        ).runSyncUnsafe()
```

Comment:
We could use `runToFuture` instead of `runSyncUnsafe` and it will work on ScalaJS too
After `s.tick()` we can check `f.value` to see if it's `Some(Failure(ex))`

Reply:
fixed that :)